### PR TITLE
Fixes #24407 - Add repository modularity API

### PR DESCRIPTION
**Pull request by johnpmitsch**
Adds endpoints for module streams

Response for `GET katello/api/v2/module_streams`

```
{
  "total": 49,
  "subtotal": 49,
  "page": 1,
  "per_page": 20,
  "error": null,
  "search": null,
  "sort": {
    "by": "name",
    "order": "asc"
  },
  "results": [
    {
      "id": 39,
      "name": "ant",
      "uuid": "db1d7b30-3db3-49c9-a141-1276dc0f8bb3",
      "version": "20180629154141",
      "context": "819b5873",
      "stream": "1.10",
      "arch": "x86_64",
      "description": "Apache Ant is a Java library and command-line tool whose mission is to drive processes described in build files as targets and extension points dependent upon each other. The main known usage of Ant is the build of Java applications. Ant supplies a number of built-in tasks allowing to compile, assemble, test and run Java applications. Ant can also be used effectively to build non Java applications, for instance C or C++ applications. More generally, Ant can be used to pilot any type of process which can be described in terms of targets and tasks.",
      "summary": "Java build tool",
      "repositories": [
        {
          "id": 9,
          "name": "rawhide modularity repo"
        }
      ]
    },
    {
      "id": 31,
      "name": "avocado",
      "uuid": "9e6a0c79-f838-40cf-b6c0-729f4ce23669",
      "version": "20180726175620",
      "context": "6c81f848",
      "stream": "latest",
      "arch": "x86_64",
      "description": "Avocado is a set of tools and libraries (what people call these days a framework) to perform automated testing.",
      "summary": "Framework with tools and libraries for Automated Testing",
      "repositories": [
        {
          "id": 9,
          "name": "rawhide modularity repo"
        }
      ]
    }
}
```

Response for `GET katello/api/v2/module_streams/39`

```
{
  "id": 39,
  "name": "ant",
  "uuid": "db1d7b30-3db3-49c9-a141-1276dc0f8bb3",
  "version": "20180629154141",
  "context": "819b5873",
  "stream": "1.10",
  "arch": "x86_64",
  "description": "Apache Ant is a Java library and command-line tool whose mission is to drive processes described in build files as targets and extension points dependent upon each other. The main known usage of Ant is the build of Java applications. Ant supplies a number of built-in tasks allowing to compile, assemble, test and run Java applications. Ant can also be used effectively to build non Java applications, for instance C or C++ applications. More generally, Ant can be used to pilot any type of process which can be described in terms of targets and tasks.",
  "summary": "Java build tool",
  "repositories": [
    {
      "id": 9,
      "name": "rawhide modularity repo"
    }
  ],
  "artifacts": [
    {
      "id": 1162,
      "name": "ant-lib-0:1.10.4-1.module_1886+ece9a977.noarch"
    },
    {
      "id": 1161,
      "name": "ant-0:1.10.4-1.module_1886+ece9a977.noarch"
    }
  ],
  "profiles": [
    {
      "id": 57,
      "name": "default",
      "rpms": [
        {
          "id": 89,
          "name": "ant"
        }
      ]
    }
  ]
}
```

### CODE REVIEW

**Comment by johnpmitsch** [comment]
Accompanying hammer PR coming soon

**Comment by johnpmitsch** [comment]
Hammer PR is here https://github.com/Katello/hammer-cli-katello/pull/576

**Comment by johnpmitsch** [comment]
[test katello]

**Comment by akofink** [comment]
Hmm...

This nonadmin user has all the filters for Products:

```yaml
$ hammer --output yaml role filters --id 17
---
- Id: 251
  Resource type: Katello::Product
  Search: none
  Unlimited?: false
  Override?: false
  Role:
    name: custom
    id: 17
    description: ''
    origin: 
  Permissions:
  - view_products
  - create_products
  - edit_products
  - destroy_products
  - sync_products
  - export_products
```

```
[ INFO 2018-08-27T17:59:47 API] GET /katello/api/module_streams              
[DEBUG 2018-08-27T17:59:47 API] Params: {                          
        "page" => 1,                                                                          
    "per_page" => 1000,                                                                                                
        "sort" => {}                                                                                   
}                                                                                                                         
[DEBUG 2018-08-27T17:59:47 API] Headers: {                                                          
    :params => {                                                                                       
            "page" => 1,                                            
        "per_page" => 1000,                                                                   
            "sort" => {}                                                                     
    }                                                                                       
}                                                                                        
[DEBUG 2018-08-27T17:59:47 API] Using authenticator: HammerCLIForeman::Api::InteractiveBasicAuth       
[ERROR 2018-08-27T17:59:48 API] 403 Forbidden                                                                                                                                                                     
[DEBUG 2018-08-27T17:59:48 API] {                                                                     
    "error" => {                                                                               
        "message" => "Access denied",                                                                  
        "details" => "Missing one of the required permissions: "                  
    }                                                                                        
}
```

I wasn't expecting a 403, and I'm surprised the error message doesn't tell me which permission I'm missing. I assume I'm missing organization view or something like that, but I'm a bit concerned about the error message. Any ideas?

**Comment by akofink** [comment]
Ok, sorry for taking so long to get back to this. It seems the permission issue is actually in the API (Postman was caching auth or something), so this requires additional permissions changes.

File: app/views/katello/api/v2/module_streams/_base.json.rabl [hunk 1]
```diff
@@ -0,0 +1,8 @@
+object @resource
```
Comment (jlsherrill) [inline app/views/katello/api/v2/module_streams/_base.json.rabl]: docker_tags is using _base, but everything else is using 'base.json.rabl'.  Id rather not propagate the minority.  usually templates starting with _ indicate its a partial (meaning not used directly), but we use our base templates directly.

**Comment by johnpmitsch** [comment]
@akofink @jlsherrill updated the permissions and added tests for the show/index permissions, also changed the '_base' rabl file to 'base'

**Comment by johnpmitsch** [comment]
@akofink I think the permissions were correct before but you have to restart the server after checking out the branch. I've updated to just have the necessary permissions

**Comment by jlsherrill** [comment]
haven't tested, but my one concern was addressed! feel free to merge based on andrew's Approval :)

**Pull request merged**